lib/pkg/flow-base/lib/FlowHelpers.js

```diff
@@ -154,38 +156,77 @@ async function canFindFlow(flowPath: string): Promise<boolean> {
   }
 }
 
+function filterJs({languageId}){
+  return languageId === 'javascript';
+}
+
+function filterCss({languageId}){
+  return languageId === 'css';
+}
+
+function findActFolder(activeDoc){
+  const docPath = activeDoc.uri.path;
+  return function({uri: {path}}){
+    return docPath.indexOf(`${path}/`) === 0;
+  };
+}
+// In case of workspace folders, calculate the root path referencing the active editor.
+function calcWorkspaceRoot(){
+  const {window: {activeTextEditor}, workspace: {rootPath, textDocuments, workspaceFolders}} = global.vscode;
+  let activeDoc, cssDocs, jsDocs, workspaceFolder, workspaceRoot;
+  if (activeTextEditor) { // If any of editors is active,
+    activeDoc = activeTextEditor.document; // target its current workspace folder.
+  } else if ((jsDocs = textDocuments.filter(filterJs)).length > 0) {
+    // otherwise, target the workspace folder accommodating the latest-opened Javascript file.
+    activeDoc = jsDocs[jsDocs.length - 1];
+  } else if ((cssDocs = textDocuments.filter(filterCss)).length > 0) {
+    // otherwise, target the workspace folder accommodating the latest-opened CSS file.
+    activeDoc = cssDocs[cssDocs.length - 1];
+  } else if (textDocuments.length > 0) {
+    // otherwise, target the workspace folder accommodating the latest-opened file.
+    activeDoc = textDocuments[textDocuments.length - 1];
+  }
+  if (activeDoc && (workspaceFolder = workspaceFolders.find(findActFolder(activeDoc)))) {
+    workspaceRoot = workspaceFolder.uri.fsPath;
+  } else {
+    workspaceRoot = rootPath;
+  }
+  return workspaceRoot;
+}
+
 /**
- * @return The path to Flow on the user's machine. First using the the user's 
+ * @return The path to Flow on the user's machine. First using the the user's
  *   config, then looking into the node_modules for the project.
- * 
- *   It is cached, so it is expected that changing the users settings will 
+ *
+ *   It is cached, so it is expected that changing the users settings will
  *   trigger a call to `clearWorkspaceCaches`.
  */
 
-async function getPathToFlow(): Promise<string> {
-  if (!global.cachedPathToFlowBin) {
-    const workspaceRoot = global.vscode.workspace.rootPath;
+function getPathToFlow(): Promise<string> {
+  const workspaceRoot = calcWorkspaceRoot();
+  if (!global.cachedPathToFlowBin[workspaceRoot]) {
+    logger.info("workspaceRoot: " + workspaceRoot);
     const config = global.vscode.workspace.getConfiguration('flow');
     const shouldUseNodeModule = config.get('useNPMPackagedFlow');
-    const userPath = config.get('pathToFlow')
-      .replace('${workspaceRoot}', workspaceRoot);
+    const userPath = config.get('pathToFlow').replace('${workspaceRoot}', workspaceRoot);
     const nodeModuleFlowPath = nodeModuleFlowLocation(workspaceRoot);
-
-    if (shouldUseNodeModule && await canFindFlow(nodeModuleFlowPath)){
-      global.cachedPathToFlowBin = nodeModuleFlowPath;
```

Comment:
Due to the lingering assignment to the cache variable "cachedPathToFlowBin", the data race causes that the path computation is executed multiple times.